autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/service/AutofillSuggestions.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2025 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.impl.service
+
+import com.duckduckgo.autofill.api.domain.app.LoginCredentials
+import com.duckduckgo.autofill.api.store.AutofillStore
+import com.duckduckgo.autofill.impl.deduper.AutofillLoginDeduplicator
+import com.duckduckgo.autofill.impl.service.mapper.AppCredentialProvider
+import com.duckduckgo.autofill.impl.ui.credential.selecting.AutofillSelectCredentialsGrouper
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+import timber.log.Timber
+
+interface AutofillSuggestions {
+    suspend fun getSiteSuggestions(website: String): List<LoginCredentials>
+    suspend fun getAppSuggestions(packageId: String): List<LoginCredentials>
+}
+
+@SingleInstanceIn(AppScope::class)
+@ContributesBinding(AppScope::class)
+class AutofillServiceSuggestions @Inject constructor(
+    private val autofillStore: AutofillStore,
+    private val loginDeduplicator: AutofillLoginDeduplicator,
+    private val grouper: AutofillSelectCredentialsGrouper,
+    private val appCredentialProvider: AppCredentialProvider,
+) : AutofillSuggestions {
+
+    override suspend fun getSiteSuggestions(website: String): List<LoginCredentials> {
```

Comment:
nit: adding `withContext(dispatchers.io())` would be good in case any future callers call this from main thread